### ISSUE TITLE
The BatchNorm error in `DataParallel`

### ISSUE THREAD

**Issue by as754770178** [pytorch/pytorch#35208]
`https://github.com/open-mmlab/mmdetection/issues/2312`
I debug this error. I set breakpoint in 
```
# /opt/anaconda3/lib/python3.7/site-packages/torch/nn/parallel/parallel_apply.py
            with torch.cuda.device(device):
                # this also avoids accidental slicing of `input` if it is a Tensor
                if not isinstance(input, (list, tuple)):
                    input = (input,)
                output = module(*input, **kwargs) # breakpoint
            with lock:
                results[i] = output
```
I found that the parameters and buffer in `cuda:1` and the input in `cuda:1`. I wonder that when the program exec to `/opt/anaconda3/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py`
The parameters and buffer of batchnorm in `cuda:0` and the input in `cuda:1`.
And only batchnorm have this error, I delete batchnorm in model. The program can run.

**Comment by SsnL** [comment]
this looks like a mmdetection bug rather than a pytorch one
